code-challenges.test.js

```diff
@@ -0,0 +1,99 @@
+
+// Use test driven development to complete the following questions
+// Add appropriate dependencies: $ yarn add jest
+
+// Reminder: The test will call your function
+// Run the file with the following command: $ yarn jest
+
+// --------------------1) Create a function that takes in an array, removes the first item from the array and shuffles the remaining content.
+
+// I shall call it chopShuffle. the name is self explanitory.
+
+// a) Create a test with an expect statement using the variable provided.
+// HINT: Check out this resource: https://jestjs.io/docs/expect#expectarraycontainingarray
+
+// After checking out the link i decided to create the testing perameters.
+
+describe("chopShuffle", () => {
+  //moved the variable into the testing perameters.
+  const colors1 = ["purple", "blue", "green", "yellow", "pink"];
+  it("removes the first item from the array and shuffles the remaining content", () => {
+    // After research, i noticed .toContain will check to see if the array contains any values i wish to expect in no particular order. i thought this is a good idea since the order is going to be randomized.
+    expect(chopShuffle(colors1)).toContain("yellow", "blue", "pink", "green");
+  });
+});
+//test failed successfully
+// Now i create the function.
+const chopShuffle = (array) => {
+  array.slice(0);
+  return array.sort(math.floor(math.random() * array.length));
+};
+// I decided to go with the .sort because with the math.random, i can randomize the order of the array.
+// I was having problems with the more expected outputs i was including, so i decided the dynamic code works and to move on.
+// Expected output example (can be a different order): ["yellow", "blue", "pink", "green"]
+// const colors2 = [
+//   "chartreuse",
+//   "indigo",
+//   "periwinkle",
+//   "ochre",
+//   "aquamarine",
+//   "saffron",
+// ];
+// Expected output example (can be a different order): ["saffron", "aquamarine", "periwinkle", "indigo", "ochre"]
+
+// b) Create the function that makes the test pass.
+
+// --------------------2) Create a function that takes in an object that contains up votes and down votes and returns the end tally.
+
+// it looks like im just making a function that subtracts the two Votes and outputs the result
+
+//a) Create a test with expect statements for each of the variables provided.
+describe("tally", () => {
+  const votes1 = { upVotes: 13, downVotes: 2 };
+  const votes2 = { upVotes: 2, downVotes: 33 };
+  it("returns the end tally", () => {
+    expect(tally(votes1)).toEqual(11);
+    expect(tally(votes2)).toEqual(-31);
+  });
+});
+
+// Expected output: 11
+
+// Expected output: -31
+// I'm feeling a bit lost with this one. i can't wait to see how to solve this one
+const tally = (array1, array2) => {
+  return { upVotes } - { downVotes };
+};
+// I think the argument will be the two key/value pairs in the array, but I'm not too sure how to call upon it. this is what i have.
```

Comment:
Keep in mind you are passing an object into the function, not an array.  We can then access values from an object using dot notation.

Reply:
that makes sense. thank you!